empower-your-google-workspace-documents-with-apps-script/content-update.js

```diff
@@ -0,0 +1,88 @@
+// Only updatable content as of 2024 is Slides and Sheets
+const driveListQuery = "(mimeType='application/vnd.google-apps.presentation' or mimeType='application/vnd.google-apps.spreadsheet') and trashed=false"
+
+// Entrypoint function
+function updateLinkedContent() {
+
+    // Turns data execution on for BigQuery data sources that are used for some Google sheets
+    SpreadsheetApp.enableBigQueryExecution()
+
+    var content = getAllFilesToBeUpdated()
+
+    // No update in Docs for any linked content
+    for (var index in content) {
+        switch (content[index].mimeType) {
+            case "application/vnd.google-apps.presentation":
+                try {
+                    let presentation = SlidesApp.openById(content[index].id)
+                    let slides = presentation.getSlides()
+                    for (var slideIndex in slides) {
+                        // Refresh linked slide
+                        if (slides[slideIndex].getSlideLinkingMode() == SlidesApp.SlideLinkingMode.LINKED) {
+                            slides[slideIndex].refreshSlide()
+                        }
+
+                        // Refresh linked sheets chart
+                        let charts = slides[slideIndex].getSheetsCharts()
+
+                        for (var chartIndex in charts) {
+                            if (charts[chartIndex].getLink() != null) {
+                                charts[chartIndex].refresh()
+                            }
+                        }
+                    }
+                    console.log(`Presentation ${content[index].id} updated successfully`);
+                }
+                catch (err) {
+                    console.log(`Presentation ${content[index].id} could not be updated: ${err.message}`);
+                }
+                break;
+            case "application/vnd.google-apps.spreadsheet":
+                try {
+                    let sheet = SpreadsheetApp.openById(content[index].id)
+                    // Only content to be updated are data sources
+                    sheet.refreshAllDataSources()
+                    console.log(`Spreadsheet ${content[index].id} updated successfully`);
+                }
+                catch (err) {
+                    console.log(`Spreadsheet ${content[index].id} could not be updated: ${err.message}`);
+                }
+                break;
+            default:
+                console.log("Unexpected mime type.")
+                break;
+        }
+    }
+}
+
+function getAllFilesToBeUpdated() {
+    // Two options:
```

Comment:
@nyanhp you can remove the comments as well as `readState()` and the `if-else` block as there is only one option

Reply:
Yup